modules/library/coverage/src/test/java/org/geotools/coverage/grid/io/ReadResolutionCalculatorTest.java

```diff
@@ -46,4 +46,34 @@ public void testReadResolutionCalculator() throws Exception {
         assertEquals(0.0331, requestedResolution[0], 1e-4);
         assertEquals(0.0331, requestedResolution[1], 1e-4);
     }
+
+    @Test
+    public void testReadResolutionCalculatorNullRes() throws Exception {
+        final CoordinateReferenceSystem requestCRS = CRS.decode("EPSG:3857", true);
+        final CoordinateReferenceSystem nativeCRS = CRS.decode("EPSG:4326", true);
+        final ReferencedEnvelope requestBounds =
+                new ReferencedEnvelope(
+                        -1.3184040176569374E7,
+                        -1.31787660216177E7,
+                        4020998.5368273035,
+                        4026272.691778981,
+                        requestCRS);
+        GridGeometry2D gg = new GridGeometry2D(new GridEnvelope2D(0, 0, 256, 256), requestBounds);
+
+        // We intentionally provide a null native res
```

Comment:
So providing a NULL native resolution is what will trigger ``ReadResolutionCalculator`` ?

Reply:
that is one case indeed.

Reply:
Nice, thx for the explanation.